### PR TITLE
[SPARK-47375][Doc][FollowUp] Fix a mistake in JDBC's preferTimestampNTZ option doc

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

Fix a mistake in JDBC's preferTimestampNTZ option doc

### Why are the changes needed?

Fix a mistake in doc

### Does this PR introduce _any_ user-facing change?

No
### How was this patch tested?

Just doc change
### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

File: docs/sql-data-sources-jdbc.md [hunk 1]
```diff
@@ -369,8 +369,8 @@ logging into the data sources.
     <td>false</td>
     <td>
       When the option is set to <code>true</code>, TIMESTAMP WITHOUT TIME ZONE type are inferred as Spark's TimestampNTZ type.
-      Otherwise, it is interpreted as Spark's Timestamp type(equivalent to TIMESTAMP WITHOUT LOCAL TIME ZONE).
-      This setting specifically affects only the inference of TIMESTAMP WITHOUT TIME ZONE data type. Both TIMESTAMP WITHOUT LOCAL TIME ZONE and TIMESTAMP WITH TIME ZONE data types are consistently interpreted as Spark's Timestamp type regardless of this setting.
+      Otherwise, it is interpreted as Spark's Timestamp type(equivalent to TIMESTAMP WITH LOCAL TIME ZONE).
```
Comment (yaooqinn) [inline docs/sql-data-sources-jdbc.md:372]: typo in L371 `are inferred` -> `is inferred`

**Review by yaooqinn — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

**Comment by yaooqinn** [comment]
It seems that we missed `TIME [W/ W/O] TZ`

**Comment by yaooqinn** [comment]
It seems that we missed `TIME [W/ W/O] TZ`

**Comment by yaooqinn** [comment]
Merged to master.

Thank you @gengliangwang @HyukjinKwon

**Comment by yaooqinn** [comment]
Added to branch-3.4/3.5 too, according https://github.com/apache/spark/pull/45502#issuecomment-1996354375

**Review by dongjoon-hyun — COMMENTED** [review]
+1, LGTM.

**Pull request closed without merging**